test/metabase/middleware_test.clj

```diff
@@ -244,39 +243,3 @@
     (test-streaming-endpoint streaming-fast-failure)
     (catch java.io.IOException e
       (.getMessage e))))
-)
-;; test that handler is killed when connection closes
-;; Note: this closing over state is a dirty hack and the whole test should be
-;; rewritten.
-
-(defn- test-slow-handler [state]
-  (fn [_]
-    (log/debug (u/format-color 'yellow "starting test-slow-handler"))
-    (reset! state :test-started)
-    (Thread/sleep 7000)  ;; this is somewhat long to make sure the keepalive polling has time to kill it.
-    (reset! state :ran-to-compleation)
-    (log/debug (u/format-color 'yellow "finished test-slow-handler"))
-    (resp/response {:success true})))
-
-(defn- start-and-maybe-kill-test-request [state kill?]
-  (reset! state :initial-state)
-  (let [path "test-slow-handler"]
-    (with-redefs [metabase.routes/routes (compojure.core/routes
-                                          (GET (str "/" path) [] (middleware/streaming-json-response
-                                                                  (test-slow-handler state))))]
-      (let  [reader (io/input-stream (str "http://localhost:" (config/config-int :mb-jetty-port) "/" path))]
-        (while-with-timeout (= :initial-state @state))
-        (when kill?
-          (.close reader))
-        (Thread/sleep 10000)))) ;; this is long enough to ensure that the handler has run to completion if it was not killed.
-  @state)
-
-;; In this first test we will close the connection before the test handler gets to change the state
-(expect
```

Comment:
These tests fail on occasion. I hit it yesterday and it hit again today, so I spent some time understanding what the test does and how it's broken.

I don't think this is a valuable test and that it's really just testing Jetty code. It uses `java.net.URL` to connect to Jetty, this is test only code.  Theres a test dummy handler that just sleeps. It then closes the stream (i.e. the client closes their connection). Jetty will then notice that and interrupt the thread servicing the request. This will throw an `InterruptedException` in the test/sleeping handler. We don't catch that exception and it just bubbles up to be swallowed by Jetty. I'm not aware of anywhere that we are setting different request thread reaping behavior or changing it's settings to warrant verifying that this Jetty behavior works.

I understand how this test works and could fix it, but it seems like wasted effort to me.